### PR TITLE
Fifth milestone -friendship V1

### PR DESCRIPTION
**Pull request by Elukoye**
This is the fifth milestone - friendship -v1 where the following was implemented:
- created friendship table
-added association between user and friendship models
-created RSpec tests for friendship model 
kindly review

### CODE REVIEW

**Comment by Elukoye** [comment]
Kindly review for us the fifth milestone in the rails final project.
The milestone includes implementing a friendship feature and writing tests to ensure the feature works.
Thanks.

**Comment by Anna-Myzukina** [comment]
> # Changes required 
> ### Hello team 
> **Very nice work on this project, however, the following changes need to be done before moving on arrow_down**
> 
> * The same error from last time is still showing up on page load, it is happening because you don't have the **comments** table's migration in your files, therefore when I clone the project and migrate I won't have a comments table and the app will break because of that, please fix that and re-submit.
> 
> ### _After making the required changes please submit a new code review request_
> ### Good luck!

Please can you run next command:    rails db: setup

File: spec/helpers/comments_helper_spec.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
+
+# Specs in this file have access to a helper object that includes
+# the CommentsHelper. For example:
+#
+# describe CommentsHelper do
```
Comment (mcrd25) [inline spec/helpers/comments_helper_spec.rb]: Please remove commented out code. 
As can be seen from [this screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/)
It is better to delete commented code. If you need to review a previous version of your code, use git history. 🙂

**Comment by Elukoye** [comment]
Kindly review the fifth milestone, we fixed the friendship model and most linter errors except one.
Please give us suggestions on how to fix it.

**Comment by Elukoye** [comment]
Hello, 
Kindly review the fifth milestone for us, we have implemented suggestions given in the last review by @Simon Wathigo

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
+class PostsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_post, only: %i[show edit update destroy]
+
+  # GET /posts
+  # GET /posts.json
+  def index
+    # @posts = Post.all
+    @posts = current_user.posts
```
Comment (maelfosso) [inline app/controllers/posts_controller.rb]: Change that line to fetch also fetch friends' post

**Comment by Elukoye** [comment]
Hello, 
Kindly review our project, we added integration tests and ensured Posts index

**Comment by Elukoye** [comment]
Hi, we covered integration tests through the controller tests kindly review

**Comment by Elukoye** [comment]
Hi, we covered integration tests through the controller tests kindly review

**Pull request merged**